app/src/main/java/com/humberto/tasky/agenda/data/task/TaskRepositoryImpl.kt

```diff
@@ -48,18 +56,31 @@ class TaskRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun deleteTask(taskId: String) {
-        val result = safeCall {
-            taskDao.deleteTask(taskId)
+    override suspend fun deleteTask(taskId: String): EmptyResult<DataError> {
+        taskDao.deleteTask(taskId)
+        return safeCall {
             agendaApi.deleteTask(taskId)
-        }
-        if(result is Result.Error) {
-            taskDao.insertModifiedTask(
-                ModifiedTaskEntity(
-                    taskId = taskId,
-                    modificationType = ModificationType.Deleted
+        }.onError { error ->
+            if (error.isRetryable()) {
+                taskDao.insertDeletedTaskSync(
+                    DeletedTaskSyncEntity(
+                        taskId = taskId,
+                        userId = localUserId!!
+                    )
```

Comment:
Same here, `applicationScope` 😄 

```
class MyApp: Application() {
    val applicationScope = CoroutineScope(SupervisorJob())
    
    // ...
}
```

Then use it to call the logic you don’t want cancelled:
```
applicationScope.launch {
    // DB call
}.join() // <— used to wait fo the coroutine to complete before returning the function
```